src/midje/data/metaconstant.clj

```diff
@@ -78,27 +80,28 @@
   clojure.lang.IPersistentCollection
   (cons [this o]
         ;; (Metaconstant. (.underlying-symbol this) (cons storage o)))
-        (throw (exceptions/user-error
-                (str "Metaconstants (" (.underlying-symbol this) ") can't have values added onto them.")
-                "If you have a compelling need for that, please create an issue:"
-                ecosystem/issues-url)))
+        (throw (report-error (str "Metaconstants (" (.underlying-symbol this) ") can't have values added onto them."))))
   (empty [this]
          (empty storage))
   (equiv [this that]
          (if (or (symbol? that)
                  (= (type that) Metaconstant)
                  (= that unbound-marker))
            (.equals this that)
-           (throw (exceptions/user-error
-                   (str "Metaconstants (" (.underlying-symbol this) ") can't be compared for equality with " (pr-str that) ".")
-                   "If you have a compelling case for equality, please create an issue:"
-                   ecosystem/issues-url))))
+           (throw
+             (report-error (str "Metaconstants (" (.underlying-symbol this) ") can't be compared for equality with "
+                                (pr-str that) ".")))))
 
   ;; Interface that provide meta support.
   clojure.lang.IObj
   (meta [this] meta-data)
   (withMeta [this m] (Metaconstant. underlying-symbol storage m)))
 
+(defn metaconstant [underlying-symbol ^clojure.lang.Associative storage meta-data]
+  (when (not (map? storage))
+    (throw (report-error (str "Metaconstants (" underlying-symbol ") can't represent non-map values " (pr-str storage) "."))))
+  (Metaconstant. underlying-symbol storage meta-data))
```

Comment:
Is this `metaconstant` function the best way to wrap the `Metaconstant.` constructor with assertion logic about the arguments?